### PR TITLE
Remove WatchRuns restriction on HectorProducer

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

- The module did nothing on Run transitions so should not restrict the framework's behavior.
- Also fixed a once per job memory leak.

#### PR validation:

Code compiles.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
NOTE: I took a quick look at the version of Hector we use in CMSSW and I found a few usages of non-const global variables. It kind of looks like those variables are meant to be set at compile time, but it is not clear. Therefore it is safest to leave HectorProducer as a `edm::one::EDProducer` (without the globals it would have been good to make it a `edm::stream::EDProducer`).

File: SimTransport/HectorProducer/src/HectorProducer.cc [hunk 1]
```diff
@@ -85,30 +82,30 @@ void HectorProducer::produce(edm::Event &iEvent, const edm::EventSetup &es) {
     throw cms::Exception("LogicError") << "HectorTrasported HepMCProduce already exists\n";
   }
 
-  evt_ = new HepMC::GenEvent(*HepMCEvt->GetEvent());
+  auto evt = new HepMC::GenEvent(*HepMCEvt->GetEvent());
```
Comment (civanch) [inline SimTransport/HectorProducer/src/HectorProducer.cc:85]: @Dr15Jones , should evt be destructed at the exit of this method?

Comment (civanch) [reply SimTransport/HectorProducer/src/HectorProducer.cc:85]: NO!

**Comment by civanch** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**